### PR TITLE
[ATL] Overload const-return-versions of FindChar/FindString/FindOneOf

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose
Fixing type problems.
JIRA issue: N/A

## Proposed changes

- Overload the non-const-return-version of `CSimpleStringT::GetString`.
- Overload the const-return-versions of `FindChar/FindCharReverse/FindString/FindOneOf`.
- Modify `CStringT::Replace`.

## TODO

- [x] Do build.

### CODE REVIEW

**Comment by JoachimHenze** [comment]
That looks like a really bad idea to me!
Why can't the return values be changed instead?
What do you want to achieve.
Please do not merge this before you have at least 3 approvals!

**Review by HBelusca — APPROVED** [review]

**Comment by JoachimHenze** [comment]
After your 2nd commits "bigFix" it looks better than before, agreed. Now less additional const_casts are needed.
But personally I will not give approval for any PR that adds any const_cast anywhere.
I do not reject though. Neutrality now.
I do recommend to wait a week before the merging. Maybe others have ideas whether it can be improved further?

File: sdk/lib/atl/cstringt.h
Comment (ThFabba) [inline sdk/lib/atl/cstringt.h:861]: To avoid the need for this, you could have two overloads of FindString -- much like wcsstr/strstr: one with const input and const return and one with non-const input and non-const return.

Comment (katahiromz) [reply sdk/lib/atl/cstringt.h]: Fixed. Thanks!

Comment (katahiromz) [reply sdk/lib/atl/cstringt.h]: Now we could fix this issue without using `const_cast`.

File: sdk/lib/atl/atlsimpstr.h
Comment (HBelusca) [inline sdk/lib/atl/atlsimpstr.h:361]: Does **MS** ATL have such const + non-const overloaded versions?

Comment (katahiromz) [reply sdk/lib/atl/atlsimpstr.h:361]: MS ATL is using const versions only:

image

Comment (katahiromz) [reply sdk/lib/atl/atlsimpstr.h:361]: And in traits:

image

Comment (HBelusca) [reply sdk/lib/atl/atlsimpstr.h:361]: Personally I think it would be just preferrable to follow the MS ATL there.

Comment (katahiromz) [reply sdk/lib/atl/atlsimpstr.h:361]: Cc @JoachimHenze

**Pull request merged**
